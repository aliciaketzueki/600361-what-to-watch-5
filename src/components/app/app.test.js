import React from "react";
import renderer from "react-test-renderer";
import App from "./app";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";

jest.mock(`../screen-main/screen-main`, () => `Main`);
jest.mock(`../screen-login/screen-login`, () => `Login`);
jest.mock(`../screen-my-list/screen-my-list`, () => `MyList`);
jest.mock(`../screen-film/screen-film`, () => `Film`);
jest.mock(`../screen-add-review/screen-add-review`, () => `AddReview`);
jest.mock(`../screen-player/screen-player`, () => `Player`);

describe(`Render App`, () => {
  it(`Render App`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <App />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -s -- -t 'Render App'
// npm run test.jest -- -u
