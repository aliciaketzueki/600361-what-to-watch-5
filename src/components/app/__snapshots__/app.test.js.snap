// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render App 1`] = `
<Main
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
/>
`;
